cmd/service-controller/definition_monitor.go

```diff
@@ -197,9 +197,21 @@ func updateAnnotatedServiceDefinition(actual *types.ServiceInterface, desired *t
 	if actual.Origin != "annotation" {
 		return false
 	}
-	if actual.Protocol != desired.Protocol || !reflect.DeepEqual(actual.Ports, desired.Ports) {
+	if actual.Protocol != desired.Protocol {
 		return true
 	}
+	if len(actual.Ports) != len(desired.Ports) {
+		return true
+	}
+	ports := map[int]int{}
+	for _, value := range desired.Ports {
+		ports[value] = value
+	}
+	for _, value := range actual.Ports {
+		if ports[value] != value {
+			return false
```

Comment:
Is it supposed to be true here?
I have looked into this again and I noticed that with false returned here, it prevents annotated services from changing ports.